src/api/interfaces/ApiResult.ts

```diff
@@ -0,0 +1,11 @@
+// standardized return format for api result
+interface ApiResult {
+	success: boolean;
```

Comment:
Question:

This is likely because of my lack of understanding of the front end. I am curious how this `success` field is used on the front end. From a perspective of service consuming the API, the HTTP status code > 200 is an indication that the result is not a success. As a result, `success` field is redundant.

But if HTTP status code is not readily available for the front end to check, having this `success` makes sense.

Reply:
This is an interesting one - the frontend currently only uses the status code to validate success or failure. I included a `success` field in anticipation of certain endpoints such as those involving bulk operations (think bulk upload themes). In such cases, the status code alone may not suffice. For example, if 10 themes were uploaded but 1 failed, is it 2XX or 4XX? In such cases, we may want to be a little more opinionated and convey our decision with the `success` attribute. It might be over-engineering things a little currently, but there might be a use case down the road.

Happy to hear of alternatives for dealing with bulk endpoints as I personally find it a little tricky. For now let's leave the field there first, we can also remove it anytime if we have other approaches 😊